src/Components/Blazor/Build/src/targets/StaticWebAssets.targets

```diff
@@ -1,37 +1,65 @@
 <Project>
 
   <PropertyGroup>
-    <ResolveStaticWebAssetsInputsDependsOn>
-      $(ResolveStaticWebAssetsInputsDependsOn);
-      _RemoveBlazorCurrentProjectAssetsFromStaticWebAssets;
-    </ResolveStaticWebAssetsInputsDependsOn>
-
-    <GetCurrentProjectStaticWebAssetsDependsOn>
-      $(GetCurrentProjectStaticWebAssetsDependsOn);
-      _RemoveBlazorCurrentProjectAssetsFromStaticWebAssets;
-    </GetCurrentProjectStaticWebAssetsDependsOn>
+    <ResolveCurrentProjectStaticWebAssetsDependsOn>
+      _ResolveBlazorGeneratedAssets;
+      $(ResolveCurrentProjectStaticWebAssetsDependsOn);
+    </ResolveCurrentProjectStaticWebAssetsDependsOn>
   </PropertyGroup>
 
-
-  <Target Name="_RemoveBlazorCurrentProjectAssetsFromStaticWebAssets">
+  <Target Name="_ResolveBlazorGeneratedAssets" DependsOnTargets="PrepareBlazorOutputs">
     <ItemGroup>
-      <StaticWebAsset Remove="@(StaticWebAsset)" Condition="'%(SourceType)' == ''" />
+      <StaticWebAsset Include="@(BlazorOutputWithTargetPath->'$(TargetDir)%(TargetOutputPath)')" RemoveMetadata="TargetOutputPath">
+        <SourceType></SourceType>
+        <SourceId>$(PackageId)</SourceId>
+        <ContentRoot>$([MSBuild]::NormalizeDirectory('$(TargetDir)\dist\'))</ContentRoot>
+        <BasePath>$(StaticWebAssetBasePath)</BasePath>
+        <RelativePath>$([System.String]::Copy('%(TargetOutputPath)').Replace('dist\','').Replace('\','/'))</RelativePath>
+      </StaticWebAsset>
+
+      <!-- We are dependingo on a "private" property for static web assets, but this is something we can clean-up in a later release.
+           These files are not "external" in the "traditional" sense but it is fine for now as this is an implementation detail. -->
+      <_ExternalStaticWebAsset Include="@(BlazorOutputWithTargetPath->'$(TargetDir)%(TargetOutputPath)')">
+        <SourceId>$(PackageId)</SourceId>
+        <!-- We just do this to keep the existing implementation happy. We will update this in the next release. -->
+        <SourceType>Generated</SourceType>
+        <ContentRoot>$([MSBuild]::NormalizeDirectory('$(TargetDir)\dist\'))</ContentRoot>
+        <BasePath>$(StaticWebAssetBasePath)</BasePath>
+      </_ExternalStaticWebAsset>
     </ItemGroup>
+
   </Target>
 
   <Target Name="BlazorStaticWebAssetsComputeFilesToPublish"
```

Comment:
What's this target trying to do?

Reply:
This is updating the publish path in the case it is a standalone app so that the assets end up published at `/publish/<app-name>/dist` as they were with the previous logic here.

I wish we could avoid all these shenanigans and just publish them to /publish/wwwroot folder in the standalone case too. That would avoid us this task alltogether.

Reply:
> I wish we could avoid all these shenanigans and just publish them to /publish/wwwroot folder in the standalone case too. That would avoid us this task alltogether.

If there isn't a technical reason we could do that, would you like to file an issue for it so we can track addressing it?

Reply:
@SteveSandersonMS How do you feel about the standalone version publishing to `publish/wwwroot` instead of `publish/<app-name>/dist`?

Reply:
I agree that would be better. But I'm fine with making that a separate change from this PR if you feel that's too disruptive right now and prefer to merge this as-is.

Reply:
I'm doing this as it simplifies things quite a bit.